scylla/README.md

```diff
@@ -0,0 +1,248 @@
+        -p scylla.hosts=ip1,ip2,ip3,... \
+        -p scylla.tokenaware=true
+
+## On choosing meaningful configuration
+
+### 1. Load target
+
+You want to test how a database handles load. To get the performance picture
+you would want to look at the latency distribution and utilization under the
+constant load. To select load use `-target` to state desired throughput level.
+
+For example `-target 120000` means that we expect YCSB workers to generate
+120,000 requests per second (RPS, QPS or TPS) to the database.
+
+Why is this important? Because without setting target throughput you will be
+looking only on the system equilibrium point that in the face of constantly
+varying latency will not allow you to see either throughput, nor latency.
+
+For more information check out these resources on the coordinated omission problem:
+
+See
+[[1]](http://highscalability.com/blog/2015/10/5/your-load-generator-is-probably-lying-to-you-take-the-red-pi.html)
+[[2]](https://medium.com/@siddontang/the-coordinated-omission-problem-in-the-benchmark-tools-5d9abef79279)
+[[3]](https://bravenewgeek.com/tag/coordinated-omission/)
+and [[this]](https://www.youtube.com/watch?v=lJ8ydIuPFeU)
+great talk by Gil Tene.
+
+### 2. Parallelism factor and threads
+
+Scylla utilizes [thread-per-core](https://www.scylladb.com/product/technology/) architecture design.
+That means that a Node consists of shards that are mapped to the CPU cores 1-per-core.
+
+In production setup, Scylla reserves 1 core to the interrupts handling and other system stuff.
+For the system with hyper threading (HT) it means 2 virtual cores. From that follows that the number
+of _Shards_ per _Node_ typically is `Number Of Cores - 2` for HT machine and
+`Number Of Cores - 1` for a machine without HT.
+
+It makes sense to select number of YCSB worker _threads_ to be multiple of the number
+of shards, and the number of nodes in the cluster. For example:
+
+    AWS Amazon i3.4xlarge has 16 vCPU (8 physical cores with HT).
+
+    =>
+
+    scylla node shards = vCPUs - 2 = 16 - 2 = 14
+
+    =>
+
+    threads = K * shards * nodes = K * 14 * nodes
+
+    for i3.4xlarge where
+
+        - K is parallelism factor >= 1,
+        - Nodes is number of nodes in the cluster.
+
+For example for 3 nodes `i3.4xlarge` and `-threads 840` means
+`K = 20`, `shards = 14`, and `threads = 14 * 20 * 3`.
+
+Thus, the `K` - the parallelism factor must be selected in the first order. If you
+don't know what you want out of it start with 1.
+
+For picking desired parallelism factor it is useful to come from desired `target`
+parameter. It is better if the `target` is a multiple of `threads`.
+
+Another concern is that for high throughput scenarios you would probably
+want to keep shards incoming queues non-empty. For that your parallelism factor
+must be at least 2.
+
+### 3. Number of connections
+
+Both `scylla.coreconnections` and `scylla.maxconnections` define limits
```

Comment:
since you use shard aware drivers
the ideal way is not to specify this at all and let the driver autodetect this based on number of shards @sitano 
however for customization purposes you can leave it around, but default for those options should be always not set (I think your client code detects that if it is not set, it will stick to autodetect)